src/librustc/mir/interpret/error.rs

```diff
@@ -296,6 +294,8 @@ pub enum InvalidProgramInfo<'tcx> {
     TypeckError,
     /// An error occurred during layout computation.
     Layout(layout::LayoutError<'tcx>),
+    /// An invalid transmute happened.
+    TransmuteSizeDiff(Ty<'tcx>, Ty<'tcx>),
```

Comment:
Given transmute checking, can't this be an ICE instead?

Reply:
It was a `bug!` once and that was actually reachable, because Miri was run on invalid MIR. Also see

https://github.com/rust-lang/rust/blob/57e1da59cd0761330b4ea8d47b16340a78eeafa9/src/librustc_mir/interpret/place.rs#L924-L929

Reply:
Oh I see, however:
1. you can't have MIR built without typeck, it just doesn't stop the compilation, and I thought `typeck_tables().has_errors` being `true` resulted in a fake MIR body
2. typeck is not at all responsible for this, the check is a MIR pass, which also doesn't stop the compilation I guess, and has no real way of indicating a transmute is bogus

And this is why we have `delay_span_bug`, for all your "this should error elsewhere, ICE if it never does" needs.

Reply:
`delay_span_bug` doesn't help, we still need to raise some error to abort interpretation.

Reply:
> `delay_span_bug` doesn't help, we still need to raise some error to abort interpretation.

Fair enough, and better to be safe than sorry.